src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -110,12 +116,56 @@ private CaseSearchResult toCaseDetailsSearchResult(MultiSearchResult multiSearch
                 throw new BadSearchRequest(errMsg);
             }
             if (response.searchResult != null) {
+                buildCaseTypesResults(response, caseTypeResults, crossCaseTypeSearchRequest);
                 caseDetails.addAll(searchResultToCaseList(response.searchResult));
                 totalHits += response.searchResult.getTotal();
             }
         }
 
-        return new CaseSearchResult(totalHits, caseDetails);
+        return new CaseSearchResult(totalHits,caseDetails,caseTypeResults);
+    }
+
+    private void buildCaseTypesResults(
+        MultiSearchResult.MultiSearchResponse response,
+        List<CaseTypeResults> caseTypeResults,
+        CrossCaseTypeSearchRequest crossCaseTypeSearchRequest) {
+        if (hitsIsNotEmpty(response)) {
+            String indexName = getIndexName(response);
```

Comment:
Been wondering about making `hitsIsNotEmpty()` and `getIndexName()` a bit more readable rather than having to search through JSON objects..

We could potentially use `List<SearchResult.Hit<ElasticSearchCaseDetailsDTO, Void>> hits = response.searchResult.getHits(ElasticSearchCaseDetailsDTO.class);` here - I think our DTO should already be suitable for that usage? Then it's much simpler:

```suggestion
        List<SearchResult.Hit<ElasticSearchCaseDetailsDTO, Void>> hits = response.searchResult.getHits(ElasticSearchCaseDetailsDTO.class);
        if (!hits.isEmpty()) {
            String indexName = hits.get(0).index;
```

Not tried this out, but looks like a good alternative if it works?

Reply:
Also - we are currently (and with my suggestion too) getting the index name from the first hit. If a service passes an ES query with `"size": 0` then there won't be any `hits` but it can still have a total - do we always expect/are we telling services that they should always pass a `size` for these kinds of queries but `_source = false`? Sorry if repeating things that have already been covered!

Reply:
It would be good,  using DTos instead of JSONs,  However  MultiSearchResult does not have any methods that access direct to getHits().   Hence  response.searchResult.getHits(ElasticSearchCaseDetailsDTO.class);

Reply:
mmm, This is a nice  one.    May be response.searchResult.getFirstHit(SearchResult.Hit.class); will be better. As we only need  to get  1) the index 2) the size and also expect empty results . Hence its is better not force a conversion to ElasticSearchCaseDetailsDTO to evaluate values from hits only.      I can test it tomorrow.  I looks possible.

Reply:
> Also - we are currently (and with my suggestion too) getting the index name from the first hit. If a service passes an ES query with `"size": 0` then there won't be any `hits` but it can still have a total - do we always expect/are we telling services that they should always pass a `size` for these kinds of queries but `_source = false`?
> 
> Sorry if repeating things that have already been covered! I can see we'll be using `_source = false` in the ticket for the original business scenario, but may need to document in known limitations so services know too?

@danlysiak this is a good catch. We did some testing and we found no way to overcome this. I think it's not a big deal because if they don't want case data to be returned, they can just use `_source:false` rather than `size:0`. I will document this limitation, cheers